### ISSUE TITLE
Doesn't appear on home.mycroft.ai

### ISSUE THREAD

**Issue by parthsuthar** [forslund/spotify-skill#27]
Installed the Skill via git to /opt/mycroft/skills (where all other skills are installed and working fine).

I can see the Podcasts configuration skill on the Skills settings page but cannot see the Spotify configurations Skill on home.mycroft.ai.

Any suggestions ?

**Comment by ConorIA** [comment]
The skill was disabled in #22 because it was hammering their servers. There is a PR that will fix/enable but isn't merged yet.

**Comment by forslund** [comment]
Hi,

The skill is currently blacklisted in mycroft due to an earlier error. The skill will be unblack-listed in thursday's release.

Until then I can only ask you to be patient

**Comment by parthsuthar** [comment]
ok @forslund @ConorIA thanks for the quick update. shall wait for the release. please keep me posted when its whitelisted.

**Comment by parthsuthar** [comment]
I did the login/authentication via home.mycroft but the agent on rpi keeps saying that its not connected and i need to go to home.mycroft and authorize. any ideas ?

**Comment by parthsuthar** [comment]
@forslund Tried it a few times now (by doing a fresh install, re-installing the skill) but still cannot authorize. Please help debug.

**Comment by parthsuthar** [comment]
@forslund @ConorIA any ideas ?

**Comment by ConorIA** [comment]
Mine showed up in the end (thought I'm have other trouble with it). @forslund will know better (I'm just a user), but I guess the first "debug steps" would be the ensure that you've updated Mycroft core and pulled the latest version of the Spotify skill.

**Comment by parthsuthar** [comment]
@ConorIA Mine also showed up. Problem is that its not connecting, despite providing the credentials and authorizing it on Spotify.

**Comment by forslund** [comment]
Hi, sorry for the delay in answering.

The first to try when debugging would be to ask `what spotify devices are available`, this will answer back one of three things:
- "This device is not connected to a spotify account" or similar asking you to go to home.mycroft.ai to authorize.
- "There are no spotify devices available. Make sure you have a Spotify Premium account", or similar if you are authorized but the skill can't find any spotify devices to play on
- A list of the devices found.

When authorizing spotify it can currently take up to 5 minutes for the changes to be applied and a connection to the account to be set up.

From what I read above I understand you get the answer that you are not authorized and should go to home to authorize the skill.

When authorized you should get a message saying: "SUCCESS! The skill is now connected to Spotify."

Does this also happen? Also you need to at least supply the spotify username for it to try to connect.

I'm going to see if I can improve the error messages somewhat tonight/tomorrow.

**Comment by parthsuthar** [comment]
I have tried and waited for more than 5 minutes but it's still not
Authorizing and I don't get the message - Success..

I have provided the correct credentials and it opens up the pop up of
Spotify to authorize but still no success :/ it keeps sending me the
message to authorize on home.mycroft.

Where else should I look ?

**Comment by forslund** [comment]
Do you not get the message success? What message do you get after you're sent to Spotify for authorization?

**Comment by parthsuthar** [comment]
Don’t get any message. It keeps showing Connect. Does the button replace to
Connected ?

**Comment by forslund** [comment]
The Button doesn't change. But clicking on it should open a pop-up taking you to the spotify page. where you authorize the skill and then it'll show the success message in the pop-up. If it doesn't do that it'll show an error message of some sort. It would be interesting to know what error message you get.

**Comment by parthsuthar** [comment]
oh okay. i do get success but its not picking up the intent. intent is handled by wikipedia. so when i search for 'play michael jackson on spotify' or even the test commands listed on your git, intent is picked up but handled by wikipedia

**Comment by parthsuthar** [comment]
even when i say ""What Spotify devices are available?"" its asks to authorize it on home.mycroft

**Comment by forslund** [comment]
Ok the `list spotify devices` intent should work to see if it's picking up any spotify devices and doesn't collide with any other intents. Make sure that one returns a device

**Comment by forslund** [comment]
Then the spotify authorization hasn't arrived yet. Restart the device and try to list devices again.

I'm out right now but later I'll do some tests to see what's happening

**Comment by parthsuthar** [comment]
ye, its not listing though i have successfully authorized it via the web (through the skill on home.mycroft).

**Comment by parthsuthar** [comment]
so finally got it working. after a couple of restarts and just waiting :) now the problem is that intent handling. when i search for `play discover weekly` it always plays this one song which isn't even part of Discover weekly. when i search for an artist, its handled by wikipedia and not spotify. raspotify is working independently - when i try to access this client from a different spotify player (on my desktop or phone).

**Comment by forslund** [comment]
Glad to hear you got it working. I set up a brand new Mycroft account yesterday evening and it all worked fine for me.

I know the search is a bit of an issue due to the colliding vocabulary. @InconsolibleCellist is working on an updated vocabulary for more options. But currently you should be able to do `play something by ARTIST` and `play the album ALBUM`.

I've never had problems with the play discover weekly. Does it play a single song or does it continue? Which song is it?

**Comment by parthsuthar** [comment]
just plays a single song from the library. doesn't accept Artist or Album. Mentioned earlier that the Artist intent is picked by Wikipedia

**Comment by forslund** [comment]
So you say `play something by david bowie` and it searches wikipedia?

Can you check the cli or send me the mycroft-skills.log file (post it on pastebin.com or similar site)

**Comment by parthsuthar** [comment]
hey @forslund so this work. when i say `play some music by david bowie` then it takes me to wikipedia but if i say `play something by david bowie` then it plays the music. so technically your skill works :) great. now to make this work -
1. which commands work - where can I look up ?
2. how do i modify/add more commands which it can pick it up and handle the intent ?

**Comment by forslund** [comment]
Great to hear that that works at least.

These are the currently supported sentences
`play PLAYLIST`
`play the album ALBUM`
`play something by ARTIST`

The vocab is being reworked in #28 to add more features like `play the album X by Y` play single song etc. Look it over and comment there if there's some missing cases you'd like to have added.

I'm closing this Issue now since it's basically working for you and the vocabulary issue is being handled elsewhere.

**Comment by estiens** [comment]
I'm still getting this. Installed the skill over an hour ago, installed fine, but doesn't show up on home.mycroft

**Comment by VHeusdens** [comment]
Same issue here, I first tried setting up my spotify account via home.mycroft.ai/skills, but although I got the 'SUCCESS! The skill is now connected to Spotify'  message my Mark1 kept saying that spotify wasn't connected and I should do so on home.mycroft.ai. In an effort to try to fix this I uninstalled- and then reinstalled - the spotify skill but now it won't show up on home.mycroft.ai/skills at all (I waited half a day and rebooted several times already, no change). It is stated as an installed skill by the mycroft-cli-client though. Is there a fix for this?
